### PR TITLE
Add C++-11 ABI check for libstdc++

### PR DESCRIPTION
**Pull request by ya1gaurav**
PR for https://github.com/open-source-parsers/jsoncpp/issues/612

### CODE REVIEW

File: src/lib_json/json_reader.cpp [hunk 1]
```diff
@@ -54,7 +54,7 @@ static size_t const stackLimit_g = JSONCPP_DEPRECATED_STACK_LIMIT; // see readVa
 
 namespace Json {
 
-#if __cplusplus >= 201103L || (defined(_CPPLIB_VER) && _CPPLIB_VER >= 520)
+#if __cplusplus >= 201103L || (defined(_CPPLIB_VER) && _CPPLIB_VER >= 520) || (_GLIBCXX_USE_CXX11_ABI > 0)
```
Comment (chfast) [inline src/lib_json/json_reader.cpp]: Definitely wrong, because adding OR condition to one I don't want to be triggered will not help.

Comment (ya1gaurav) [inline src/lib_json/json_reader.cpp]: Replaced with AND

File: src/lib_json/json_reader.cpp [hunk 2]
```diff
@@ -54,7 +54,7 @@ static size_t const stackLimit_g = JSONCPP_DEPRECATED_STACK_LIMIT; // see readVa
 
 namespace Json {
 
-#if __cplusplus >= 201103L || (defined(_CPPLIB_VER) && _CPPLIB_VER >= 520)
+#if __cplusplus >= 201103L || (defined(_CPPLIB_VER) && _CPPLIB_VER >= 520) && (_GLIBCXX_USE_CXX11_ABI > 0)
```
Comment (chfast) [inline src/lib_json/json_reader.cpp]: WTF? Now I have `defined(_CPPLIB_VER) && _CPPLIB_VER >= 520 && _GLIBCXX_USE_CXX11_ABI > 0` what is never the case. I don't have time for this.

**Comment by timofonic** [comment]
Hello @chfast

Why this and other pull requests are open instead of merged or closed?

Kind regards.

**Comment by chfast** [comment]
I don't know. I'm not the owner of this project.

I think the changes in this PR does not have much sense.

**Pull request closed without merging**